challenge.rb

```diff
@@ -1,26 +1,38 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  array = Array.new(input)
+
+  array.map! { |i| i.capitalize }
+  return array
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  array = Array.new(input)
+  array.keep_if{ |i| i == 'dog'}
+  return array
```

Comment:
You don't need to explicitly ```return array``` here, it's sufficient to just say ```array``` and ruby will return the last call